### DISCUSSION TITLE
Firebase Authentication for Android Credential Manager

### DISCUSSION THREAD

**Discussion by br0ll** [capawesome-team/capacitor-firebase#740] [General]
Hey! While trying to fix an issue with app switching for authentication, I stumbled upon the deprectation of `GoogleSignInOptions` and related classes, in favor of the new Credential Manager API (https://developer.android.com/identity/sign-in/credential-manager-siwg).

I'll try to implement this in my project switching fork.

**Comment by robingenz** [comment]
Would you be willing to create a PR? I hope there are not too many changes. I can't keep up with the reviews at the moment. 😅

**Reply by br0ll** [reply]
I'll try! It should only affect the GoogleAuthProviderHandler of the Firebase Authentication plugin, so hopefully noy too many changes, no :)

**Reply by davidgalvis-sicpa** [reply]
Indeed, I've received this on my Google PlayStore inbox:

You're utilizing legacy Google Sign-in APIs, which are deprecated and planned to be removed in 2025. For details on migrating to Sign in with Google via Credential Manager, read our [migration guide](https://notifications.google.com/g/p/ANiao5qiBZje4nLees_NFAJMuyZnHBmzmWdH4aC1a_kJVCsta0WqE2GvsJ_Hl6fmUBJX7Y1pp-j4NfXITGpeuaQ7xPDAuJC0uo0U8wFaKNSoSyfEmJTEWqFfPt3RQsNXIrtRTrApr0-Bp9SkZ_aTCvKQcgYIGTbRtGRK6BtqE6vKM7q6uSNUzciMcyoUotFhygsHfa4Hpa2Ew53pL2nIw_A-Mzd1ZYiBagpb9fHJq5zQnXQO9OnhhMbbuUdEFpm-FqMkumo_yPGfrJR0raQp-RC4JiLoJ-iJMmm2).

Any ideas on there this will be updated on capacitor-firebase?

**Reply by robingenz** [reply]
Yes, we will work on this.
